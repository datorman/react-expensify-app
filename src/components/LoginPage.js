import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from './../actions/auth';

export const LoginPage = ({startLogin}) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className = "box-layout__title">Expensify Login</h1>
            <p>Manage Your Expenses</p>
            <button className = "button box-layout__button" onClick = {startLogin}>Login With Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) =>({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);