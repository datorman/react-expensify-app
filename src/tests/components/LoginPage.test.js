import React from 'react';
import LoginPage from './../../components/LoginPage';
import {shallow} from 'enzyme';

let wrapper, startLogin;

beforeEach(()=>{
    wrapper = shallow(<LoginPage/>)
});

test('Should render Login page', () =>{
    expect(wrapper).toMatchSnapshot();
})
test('Should call startLogin on button click', () =>{

});