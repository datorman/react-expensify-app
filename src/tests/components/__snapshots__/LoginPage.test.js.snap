// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Login page 1`] = `undefined`;
